src/python/pants/backend/java/target_types.py

```diff
@@ -142,6 +144,40 @@ async def generate_targets_from_java_sources(
     )
 
 
+# Things for JARs
+#
+
+
+class JvmMainClassName(StringField):
+    alias = "main"
+    required = True
+    help = (
+        "`.`-separated name of the JVM class containing the `main()` method to be called when "
+        "executing this JAR."
+    )
+
+
+class JvmRootClassAddress(StringField):
+    alias = "root_address"
+    required = True
+    help = "Address that contains the Java source which contains the `main` class."
+
+
+class FatJar(Target):
+    alias = "fat_jar"
+    core_fields = (
+        *COMMON_TARGET_FIELDS,
+        Dependencies,
+        OutputPathField,
+        JvmMainClassName,
+        JvmRootClassAddress,
+    )
+    help = (
+        "A `jar` file that continains the compiled source code, as well as all of its "
+        "dependencies."
```

Comment:
also mention that the dependencies are flattened into the jar.

Reply:
(since for example a `.war` file technically also contains its dependencies)

Reply:
I went with the following, to be super clear about the resulting structure ("flattening" sounds like making a flat directory structure, too, which is not what's going on here)

```
        "A `jar` file that continains the compiled source code along with its dependency class "
        "files, where the compiled class files from all dependency JARs, along with first-party "
        "class files, exist in a common directory structure."

```